src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ExtensionMethodImportCompletionHelper.cs

```diff
@@ -0,0 +1,483 @@
+
+        public static async Task<(ImmutableArray<SerializableImportCompletionItem>, StatisticCounter)> GetUnimportedExtensionMethodsInCurrentProcessAsync(
+            Document document,
+            int position,
+            ITypeSymbol receiverTypeSymbol,
+            ISet<string> namespaceInScope,
+            bool forceIndexCreation,
+            CancellationToken cancellationToken)
+        {
+            var counter = new StatisticCounter();
+            var ticks = Environment.TickCount;
+
+            var project = document.Project;
+            var assembly = (await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false))!;
+
+            // Get the metadata name of all the base types and interfaces this type derived from.
+            using var _ = PooledHashSet<string>.GetInstance(out var allTypeNamesBuilder);
+            allTypeNamesBuilder.Add(receiverTypeSymbol.MetadataName);
+            allTypeNamesBuilder.AddRange(receiverTypeSymbol.GetBaseTypes().Select(t => t.MetadataName));
+            allTypeNamesBuilder.AddRange(receiverTypeSymbol.GetAllInterfacesIncludingThis().Select(t => t.MetadataName));
+
+            // interface doesn't inherit from object, but is implicitly convertable to object type.
+            if (receiverTypeSymbol.IsInterfaceType())
+            {
+                allTypeNamesBuilder.Add(nameof(Object));
+            }
+
+            var allTypeNames = allTypeNamesBuilder.ToImmutableArray();
+            var matchedMethods = await GetPossibleExtensionMethodMatchesAsync(
+                project, allTypeNames, forceIndexCreation, isPrecalculation: false, cancellationToken).ConfigureAwait(false);
+
+            counter.GetFilterTicks = Environment.TickCount - ticks;
+            counter.NoFilter = matchedMethods == null;
+
+            // Don't show unimported extension methods if the index isn't ready.
+            if (matchedMethods == null)
+            {
+                // We use a very simple approach to build the cache in the background:
+                // queue a new task only if the previous task is completed, regardless of what
+                // that task is.
+                lock (s_gate)
+                {
+                    if (s_indexingTask.IsCompleted)
+                    {
+                        s_indexingTask = Task.Run(() => GetPossibleExtensionMethodMatchesAsync(project, allTypeNames, forceIndexCreation: false, isPrecalculation: true, CancellationToken.None));
+                    }
+                }
+
+                return (ImmutableArray<SerializableImportCompletionItem>.Empty, counter);
+            }
+
+            ticks = Environment.TickCount;
+
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var items = GetExtensionMethodItems(assembly.GlobalNamespace, receiverTypeSymbol,
+                semanticModel!, position, namespaceInScope, matchedMethods, counter, cancellationToken);
```

Comment:
i don't see how this could work.  here's the issue:

1. you looked at all dependent projects to add their pe references  and get the indices for them.
2. you're now walking your particular assembly.GlobalNamespace symbol actually looking for the extension methods.

This would not find any extension methods for pe references that this compilation doesn't itself point at.  

So basically, i just don't get how this coudl work.

Reply:
As discussed offline, transitive PE reference isn't support in C#/VB, so this is simply a mistake and I will remove related code.